### PR TITLE
PGOV-1234 Add a weekly re-sync job to retrieve user info from Entra ID

### PR DESCRIPTION
**Pull request by kxwang**

### CODE REVIEW

File: web/modules/custom/portland/modules/portland_openid_connect/portland_openid_connect.module [hunk 1]
```diff
@@ -193,44 +97,64 @@ function _get_all_users($access_token)
...
+    $domain = "portlandoregon.gov";
+    if ($resume_url = $system_state->get("pgov.user_sync.resume_url.$domain")) {
+      \Drupal::logger('portland OpenID')->notice("Resuming user sync for $domain");
     } else {
-      \Drupal::logger('portland OpenID')->notice('Start retrieving user information ...');
+      \Drupal::logger('portland OpenID')->notice("Starting user sync for $domain");
     }
 
+    // Get the user sync batch size. Default is 50 users
+    $batch_size = \Drupal::state()->get('pgov.user_sync.batch_size') ?? "100";
```
Comment (RichardDavies) [inline web/modules/custom/portland/modules/portland_openid_connect/portland_openid_connect.module:115]: Is the comment above incorrect? Should it say 100 users?

**Review by RichardDavies — DISMISSED** [review]
Approved, but one possible edge case that may deserve consideration.

File: web/modules/custom/portland/modules/portland_openid_connect/portland_openid_connect.module (hunk 1 above)
Comment (kxwang) [reply web/modules/custom/portland/modules/portland_openid_connect/portland_openid_connect.module]: Comment updated

File: web/modules/custom/portland/modules/portland_openid_connect/portland_openid_connect.module [hunk 2]
```diff
@@ -75,13 +85,51 @@ function portland_openid_connect_cron()
...
+    "access_token" => $access_token,
+    "domain" => $domain,
+    "user_type" => "drupal",
+    "users" => $users
+  ]);
+
+  // Update the user offset
+  \Drupal::state()->set('pgov.user_sync.drupal_user_offset', $drupal_user_offset + count($uids));
+  // Done when the returned user count is less than the limit
+  if (count($uids) < $drupal_user_limit) {
```
[the remark this answers is not in the document]
Comment (kxwang) [reply web/modules/custom/portland/modules/portland_openid_connect/portland_openid_connect.module:129]: Good point. I've added an IF statement around running the user query.

**Pull request merged**